packages/react/src/components/uiComponents.tsx

```diff
@@ -276,7 +234,19 @@ const UserButtonContext = createContext<MountProps>({
 });
 
 const _UserButton = withClerk(
-  ({ clerk, ...props }: WithClerkProp<PropsWithChildren<UserButtonPropsWithoutCustomPages>>) => {
+  ({
+    clerk,
+    component,
+    fallback,
+    ...props
+  }: WithClerkProp<PropsWithChildren<UserButtonPropsWithoutCustomPages> & FallbackProp>) => {
+    const mountingStatus = useWaitForComponentMount(component);
+    const shouldShowFallback = mountingStatus === 'rendering' || !clerk.loaded;
+
+    const rendererRootProps = {
+      ...(shouldShowFallback && fallback && { style: { display: 'none' } }),
```

Comment:
@panteliselef whats the reasoning?

Reply:
styling wise yes, it's kind of nitpicking tbh, but afiak screen readers don't respect `display:none;` but respect `hidden`.

Reply:
Got it, I think it should be alright, since its just hiding the container for the components to be rendered into until the components have loaded. The loading text sits outside of it, so it is accessible to users.

<img width="821" alt="loading" src="https://github.com/user-attachments/assets/bf6348f4-1a3a-4e0f-b3ff-0a4d244f931f">